gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -322,8 +339,69 @@ gst_tensor_src_iio_init (GstTensorSrcIIO * self)
   self->silent = DEFAULT_PROP_SILENT;
   self->buffer_capacity = DEFAULT_BUFFER_CAPACITY;
   self->sampling_frequency = DEFAULT_FREQUENCY;
+  self->merge_channels_data = DEFAULT_MERGE_CHANNELS;
+  self->is_tensor = FALSE;
+  self->config.tensors = NULL;
+
+  /**
+   * format of the source since IIO device as a source is live and operates
+   * at a fixed frequency, GST_FORMAT_TIME is used
+   */
+  gst_base_src_set_format (GST_BASE_SRC (self), GST_FORMAT_TIME);
+  /** set the source to be live */
+  gst_base_src_set_live (GST_BASE_SRC (self), TRUE);
+}
+
+/**
+ * @brief merge multiple other/tensor
+ * @note they should have matching type and shape to form 1 other/tensors
+ * @note extra dimension should be available for other/tensors
+ * @note order of merge is stable
+ * @param[in/out] info Tensor info to be merged
+ * @param[in] size Info array size
+ * @returns >=0 number of valid entries in the info after merge
+ *         -1 failed due to missing extra dimension/mismatch shape/type
+ * @todo merge tensors (in stable order) into buckets based on type
+ */
+static gint
+gst_tensor_src_merge_tensor_by_type (GstTensorInfo * info, guint size)
+{
+  gint info_idx, base_idx, dim_idx;
+  gboolean mismatch = FALSE, dim_avail = TRUE;
+  gint ret = 0;
```

Comment:
I think ret is unnecessary in this function

Reply:
Updated.
Also added graceful handling of failed merging.